Android/Shared/src/main/java/com/bentley/sample/shared/SampleITMApplication.kt

```diff
@@ -77,6 +79,44 @@ open class SampleITMApplication(context: Context, attachWebViewLogger: Boolean,
                 throw ex
             }
         }
+
+        coMessenger.registerMessageHandler("firstRenderStarted") {
+            logger.log(ITMLogger.Severity.Debug, "Received firstRenderStarted")
+        }
+        coMessenger.registerMessageHandler("firstRenderFinished") {
+            logger.log(ITMLogger.Severity.Debug, "Received firstRenderFinished")
+        }
+    }
+
+    /**
+     * Checks for values in configData with a prefix of "ITMSAMPLE_ACTION_" and constructs a [JsonObject]
+     * using everything after ITMSAMPLE_ACTION_ as the key and the values from configData.
+     * @return A [JsonObject] containing all the ITMSAMPLE_ACTION_ prefixed values from configData.
+     */
+    protected open fun getActionsFromConfigData(): JsonObject {
```

Comment:
The `if` statements with the `continue` call seem unnecessary to me.

I think this is easier to read and more idiomatic:

```kotlin
    protected open fun getActionsFromConfigData(): JsonObject {
        val actions = JsonObject()
        configData?.let { configData ->
            for (key in (configData.names())) {
                val shortKey = key.removePrefix("ITMSAMPLE_ACTION_")
                if (shortKey.length <= key.length) {
                    configData.getOptionalString(key)?.let { value ->
                        actions[shortKey] = value
                    }
                }
            }
        }
        return actions
    }
```

Reply:
Sorry about that `<=` typo, that was obviously wrong.

If you want to match the Swift code, you can actually iterate over the configData entries like this:
`for (entry in configData)`

Unfortunately, destructuring doesn't work for `JSONObject` though, so you would still have to introduce local vals or use entry.key and entry.value.

Lastly, my code also had extra unnecessary parens around `configData.names()`.

Reply:
I removed the parens. I left the enumeration alone, since I'm not trying to exactly match the two sets of code, just have them be similar while retaining the idioms of each language.